src/bsp/platform/iotc_bsp_io_net_test.cc

```diff
@@ -0,0 +1,128 @@
+/* Copyright 2018-2019 Google LLC
+ *
+ * This is part of the Google Cloud IoT Device SDK for Embedded C,
+ * it is licensed under the BSD 3-Clause license; you may not use this file
+ * except in compliance with the License.
+ *
+ * You may obtain a copy of the License at:
+ *  https://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "iotc_test_echoserver.h"
+#include "gtest.h"
+#include "iotc_bsp_io_net.h"
+
+#include <stdio.h>
+
+namespace iotctest {
+namespace {
+
+typedef struct NetworkTestCase {
+  iotc_bsp_socket_type_t socket_type;
+  iotc_bsp_protocol_type_t protocol_type;
+  char* listening_addr;
```

Comment:
`std::string listening_addr;` ?

That would make your life easier when you create the test cases in `INSTANTIATE_TEST_CASE_P`, and you have access to the underlying data with `.c_str()`, to pass to `iotc_bsp_io_net_socket_*`.